### PR TITLE
added styled components

### PR DESCRIPTION
**Pull request by dmtryiudin**
added styled components

closing issue https://github.com/igorizviekov/mt-booking_app/issues/27

### CODE REVIEW

File: src/components/ButtonComponent/buttonComponent.types.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+/**
+ * Types for Button
+ * @param {primary} boolean - primary style of the button
+ * @param {secondary} boolean - secondary style of the button
+ */
+
+export interface IButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  primary?: boolean;
+  secondary?: boolean;
+}
```
Comment (igorizviekov) [inline src/components/ButtonComponent/buttonComponent.types.d.ts:12]: 1. why do you need 2 flags for one thing in your interface?
2. Replace `primary` flag with `type`

File: src/sections/Home/Home.tsx [hunk 2]
```diff
@@ -26,6 +27,8 @@ export const Home = () => {
         cupiditate, expedita neque eius reiciendis in iusto natus saepe, culpa
         voluptas nostrum, id quo unde odit facilis corporis minima amet.
       </article>
+      <ButtonComponent primary>Primary</ButtonComponent>
```
Comment (igorizviekov) [inline src/sections/Home/Home.tsx:30]: Please refactor all the components you have not to include the word "Component". This makes your code less readable

**Review by igorizviekov — CHANGES_REQUESTED** [review]

File: src/components/ButtonComponent/ButtonComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import styled, { css } from 'styled-components';
+import { IButtonProps } from './buttonComponent.types';
+import React from 'react';
+
+const Button = styled.button<{ primary?: boolean; secondary?: boolean }>`
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/ButtonComponent/ButtonComponent.tsx]: https://github.com/igorizviekov/mt-booking_app/commit/9e30f23436e309c0adfecc5cea70b3606e6bef70

File: src/components/ButtonComponent/buttonComponent.types.d.ts (hunk 1 above)
Comment (ufoBeliever) [reply src/components/ButtonComponent/buttonComponent.types.d.ts:22]: https://github.com/igorizviekov/mt-booking_app/commit/3e113f3ad147eb58ef2301f0ddc6984a2246951c

File: src/sections/Home/Home.tsx (hunk 2 above)
Comment (ufoBeliever) [reply src/sections/Home/Home.tsx]: https://github.com/igorizviekov/mt-booking_app/commit/77033a80d298d75278149311ba8b4af03bc0c71d

File: src/components/ButtonComponent/ButtonComponent.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (igorizviekov) [reply src/components/ButtonComponent/ButtonComponent.tsx]: please refactor it to use relevant props

File: src/components/ButtonComponent/ButtonComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  margin: 0 1em;
+  padding: 0.25em 1em;
+
+  ${(props) =>
+    props.primary &&
+    css`
+      background: palevioletred;
+      color: white;
+    `};
+`;
```
[the remark this answers is not in the document]
Comment (igorizviekov) [reply src/components/ButtonComponent/ButtonComponent.tsx]: 1. Just remove it to keep it clean
2. Pelase refactor it to import whole styled button const from the file where you keep the styles and use it like so `<StyledButton />`

File: src/components/Button/button.types.d.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Types for Styled Button
+ * @param {primary} boolean - primary style of the button
+ * @param {secondary} boolean - secondary style of the button
+ */
+
+export interface IStyledButton {
+  primary?: boolean;
+  secondary?: boolean;
+}
```
Comment (igorizviekov) [inline src/components/Button/button.types.d.ts:22]: why do we need a separate interface for a styled button? Can it use the `IButtonProps` ?

Comment (ufoBeliever) [reply src/components/Button/button.types.d.ts:10]: https://github.com/igorizviekov/mt-booking_app/pull/30/commits/44608ab65a6e86668930ffc2be6f022cec2a0cd8
https://github.com/igorizviekov/mt-booking_app/pull/30/commits/2f563bbead9993992fe9f7565954d4fd84e79f2b

Comment (ufoBeliever) [reply src/components/Button/button.types.d.ts:10]: https://github.com/igorizviekov/mt-booking_app/pull/30/commits/44608ab65a6e86668930ffc2be6f022cec2a0cd8
https://github.com/igorizviekov/mt-booking_app/pull/30/commits/2f563bbead9993992fe9f7565954d4fd84e79f2b

File: src/components/Button/button.types.d.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { HTMLAttributes } from 'react';
+
+/**
+ * Types for Button
+ * @param {styleType} string - set or secondary primary style of the button
+ */
+
+export interface IButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  styleType?: 'primary' | 'secondary';
```
Comment (igorizviekov) [inline src/components/Button/button.types.d.ts:9]: I think that `styleType` is a bad naming for the prop as in the future the behavior of the btn can change, not only its styles. So refactor it to be just `type`.

**Review by igorizviekov — CHANGES_REQUESTED** [review]

File: src/components/Button/button.types.d.ts (hunk 5 above)
Comment (igorizviekov) [reply src/components/Button/button.types.d.ts:10]: one more commit needed please =)

**Review by igorizviekov — APPROVED** [review]

**Pull request merged**